torch/_inductor/compile_fx.py

```diff
@@ -1344,6 +1364,15 @@ def partition_fn(graph, joint_inputs, **kwargs):
     @dynamo_utils.dynamo_timed
     @dynamo_utils.maybe_cprofile
     def bw_compiler(model: torch.fx.GraphModule, example_inputs: List[torch.Tensor]):
+        user_visible_outputs = set()
+
+        if config.bw_outputs_user_visible:
+            *_, model_outputs_node = model.graph.nodes
+            assert model_outputs_node.op == "output"
+            model_outputs = pytree.arg_tree_leaves(*model_outputs_node.args)
+            user_visible_outputs = {
+                n.name for n in model_outputs if isinstance(n, torch.fx.Node)
+            }
```

Comment:
sets have non-determinstic order, which can lead to annoying issues. Suggest `dict.fromkeys`